### PR TITLE
Speed up extraction mode

### PR DESCRIPTION
**Pull request by VashishtMadhavan**
This PR speeds up extraction mode by:
* Converting to async calls
* Retrieving the 100 most relevant docs to limit number of calls

TODO:
* Note the actual speed up from async
* Aggregate and dedup extraction results by document

### CODE REVIEW

File: models/extractor.py [hunk 1]
```diff
@@ -89,12 +92,21 @@ def generate_csv(self, entities: Dict, results: List[Any]):
...
+        return await asyncio.gather(*tasks)
+
     def run(self, docs: List[Document]) -> List[ExtractionResult]:
         output = []
-        for doc in docs:
-            question = self._get_question()
-            result = self.chain.run(context=doc.page_content, question=question)
+        results = asyncio.run(self.async_run_docs(docs, self.question))
 
+        for result, doc in zip(results, docs, strict=True):
```
Comment (andylouisqin) [inline models/extractor.py]: Interesting, `strict` was [introduced](https://peps.python.org/pep-0618/) in python 3.10 and I'm on 3.9.x. Is there a way to assert a python version for a project besides the readme?

File: main.py [hunk 2]
```diff
@@ -75,7 +75,7 @@ def search(query: str) -> Dict[str, Any]:
 
 
 @app.get("/extract")
-def extract(entity_json: str) -> Dict[str, Any]:
+def extract(entity_json: str) -> StreamingResponse:
```
Comment (andylouisqin) [inline main.py:78]: Good catch. Seems like pydantic doesn't really typecheck if it was allowed to be wrong in the first place?

**Review by andylouisqin — DISMISSED** [review]
This runs so much faster; well done. We might want a configurable `k` in the UI later, but we'll take this huge speedup for now

File: main.py (hunk 2 above)
Comment (VashishtMadhavan) [reply main.py:78]: It only type checks when you run it with a linter. I used the `make format` & `make lint` the readme.

File: models/extractor.py (hunk 1 above)
Comment (VashishtMadhavan) [reply models/extractor.py]: Ah gotcha, let me remove to support back compat

Comment (VashishtMadhavan) [reply models/extractor.py]: Idk if there is easy way to python version check. Lets revisit later potentially

**Pull request merged**